### ISSUE TITLE
nodejs/iojs fails to build with libreSSL

### ISSUE THREAD

**Issue by spacekitteh** [NixOS/nixpkgs#20499]
Upstream issue: https://github.com/nodejs/node/issues/1622

Here is how Gentoo fixed it for an old version: https://github.com/gentoo/libressl/pull/117/files

Upstream thread about migrating to libreSSL: https://github.com/nodejs/node/issues/428#issuecomment-228494226

Upstream WIP PR about migrating: https://github.com/nodejs/node/pull/9376

**Comment by qolii** [comment]
It is, I think. I tried building `nodejs-8_x` with `libressl` and got a build failure around the SSL parts.

I also tried building `nodejs-10_x` and saw that it somehow didn't take my override, but pulled in actual `openssl`. I'm not entirely clear on how it managed that, but the build appeared to complete fine in this case.

In any case, looking at the upstream `nodejs` Issue tracker, `libressl` seems to still be very much unsolved.
